src/network/gun/version.2/post.ts

```diff
@@ -0,0 +1,79 @@
+import { OnlyRunInContext } from '@holoflows/kit/es'
+import { gun2, PostOnGun2, SharedAESKeyGun2 } from '.'
+import { hashPostSalt, hashCryptoKey } from './hash'
+import { PublishedAESKeyRecordV39 } from '../../../crypto/crypto-alpha-39'
+
+OnlyRunInContext('background', 'gun')
+
+/**
+ * Query all possible keys stored on the gun
+ * @param postSalt Post iv
+ * @param partitionByCryptoKey Public key of the current user (receiver)
+ */
+export async function queryPostKeysOnGun2(
+    postSalt: string,
+    partitionByCryptoKey: CryptoKey,
+): Promise<SharedAESKeyGun2[]> {
+    const postHash = await hashPostSalt(postSalt)
+    const keyHash = await hashCryptoKey(partitionByCryptoKey)
+
+    // ? here we get the internal node names of gun2[postHash][keyHash]
+    // ? where gun2[postHash][keyHash] is a list
+    const internalNodes: Record<string, string> =
+        (await gun2
+            .get(postHash)
+            // @ts-ignore
+            .get(keyHash)
+            .once().then!()) || {}
+    // ? In this step we get something like ["jzarhbyjtexiE7aB1DvQ", "jzarhuse6xlTAtblKRx9"]
+    const internalKeys = Object.keys(internalNodes).filter(x => x !== '_')
+    // ? In this step we get all keys in this category (gun2[postHash][keyHash])
+    const resultPromise = internalKeys.map(key => gun2.get(key).once().then!())
+    const result = (await Promise.all(resultPromise)) as SharedAESKeyGun2[]
+    return result
+}
+
+/**
+ * Listen on the changes of all possible keys on the gun
+ * @param postSalt Post iv
+ * @param partitionByCryptoKey Public key of the current user (receiver)
+ * @param callback
+ */
+export function subscribePostKeysOnGun2(
+    postSalt: string,
+    partitionByCryptoKey: CryptoKey,
+    callback: (data: PostOnGun2) => void,
+) {
+    hashPostSalt(postSalt).then(postHash => {
+        hashCryptoKey(partitionByCryptoKey).then(keyHash => {
+            gun2.get(postHash)
+                // @ts-ignore
+                .get(keyHash)
+                .map()
+                .on((data: PostOnGun2) => {
+                    const data2 = Object.assign({}, data)
+                    delete data2._
+                    callback(data2)
+                })
+        })
+    })
+    // tslint:disable-next-line: no-parameter-reassignment
+    return () => (callback = () => {})
```

Comment:
what does this (and this function) do...

Reply:
I want to remove the listener but the gun's `off` has bugs. 
This line replaces the original `callback` and this new `callback` doesn't hold any reference at the current closure so no memory will be leaked.